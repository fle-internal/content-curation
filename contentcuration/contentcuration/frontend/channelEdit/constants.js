import { fileErrors } from 'shared/constants';

export const RouterNames = {
  SANDBOX: 'SANDBOX',
  TREE_ROOT_VIEW: 'TREE_ROOT_VIEW',
  TREE_VIEW: 'TREE_VIEW',
  STAGING_TREE_VIEW: 'STAGING_TREE_VIEW',
  CONTENTNODE_DETAILS: 'CONTENTNODE_DETAILS',
  MULTI_CONTENTNODE_DETAILS: 'MULTI_CONTENTNODE_DETAILS',
  IMPORT_FROM_CHANNELS: 'IMPORT_FROM_CHANNELS',
  IMPORT_FROM_CHANNELS_BROWSE: 'IMPORT_FROM_CHANNELS_BROWSE',
  IMPORT_FROM_CHANNELS_SEARCH: 'IMPORT_FROM_CHANNELS_SEARCH',
  IMPORT_FROM_CHANNELS_REVIEW: 'IMPORT_FROM_CHANNELS_REVIEW',
  ADD_TOPICS: 'ADD_TOPICS',
  ADD_EXERCISE: 'ADD_EXERCISE',
  UPLOAD_FILES: 'UPLOAD_FILES',
  TRASH: 'TRASH',
  ADD_PREVIOUS_STEPS: 'ADD_PREVIOUS_STEPS',
  ADD_NEXT_STEPS: 'ADD_NEXT_STEPS',
};

export const viewModes = {
  DEFAULT: 'DEFAULT_VIEW',
  COMFORTABLE: 'COMFORTABLE_VIEW',
  COMPACT: 'COMPACT_VIEW',
};

export const ValidationErrors = {
  TITLE_REQUIRED: 'TITLE_REQUIRED',
  LICENCE_REQUIRED: 'LICENCE_REQUIRED',
  COPYRIGHT_HOLDER_REQUIRED: 'COPYRIGHT_HOLDER_REQUIRED',
  LICENCE_DESCRIPTION_REQUIRED: 'LICENCE_DESCRIPTION_REQUIRED',
  MASTERY_MODEL_REQUIRED: 'MASTERY_MODEL_REQUIRED',
  MASTERY_MODEL_INVALID: 'MASTERY_MODEL_INVALID',
  QUESTION_REQUIRED: 'QUESTION_REQUIRED',
  INVALID_NUMBER_OF_CORRECT_ANSWERS: 'INVALID_NUMBER_OF_CORRECT_ANSWERS',
  NO_VALID_PRIMARY_FILES: 'NO_VALID_PRIMARY_FILES',
  ...fileErrors,
};

// should correspond to backend types
export const AssessmentItemTypes = {
  SINGLE_SELECTION: 'single_selection',
  MULTIPLE_SELECTION: 'multiple_selection',
  TRUE_FALSE: 'true_false',
  INPUT_QUESTION: 'input_question',
};

export const AssessmentItemToolbarActions = {
  EDIT_ITEM: 'EDIT_ITEM',
  MOVE_ITEM_UP: 'MOVE_ITEM_UP',
  MOVE_ITEM_DOWN: 'MOVE_ITEM_DOWN',
  DELETE_ITEM: 'DELETE_ITEM',
  ADD_ITEM_ABOVE: 'ADD_ITEM_ABOVE',
  ADD_ITEM_BELOW: 'ADD_ITEM_BELOW',
};

export const AssessmentItemTypeLabels = {
  [AssessmentItemTypes.SINGLE_SELECTION]: 'questionTypeSingleSelection',
  [AssessmentItemTypes.MULTIPLE_SELECTION]: 'questionTypeMultipleSelection',
  [AssessmentItemTypes.TRUE_FALSE]: 'questionTypeTrueFalse',
  [AssessmentItemTypes.INPUT_QUESTION]: 'questionTypeInput',
};

export const TabNames = {
  DETAILS: 'details',
  PREVIEW: 'preview',
  QUESTIONS: 'questions',
  RELATED: 'related',
};

export const modes = {
  EDIT: 'EDIT',
  NEW_TOPIC: 'NEW_TOPIC',
  NEW_EXERCISE: 'NEW_EXERCISE',
  UPLOAD: 'UPLOAD',
  VIEW_ONLY: 'VIEW_ONLY',
};
