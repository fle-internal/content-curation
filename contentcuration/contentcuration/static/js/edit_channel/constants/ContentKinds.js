module.exports = [
  {
    kind: 'document',
  },
  {
    kind: 'topic',
  },
  {
    kind: 'video',
  },
  {
    kind: 'html5',
  },
  {
    kind: 'audio',
  },
  {
    kind: 'exercise',
  },
];
