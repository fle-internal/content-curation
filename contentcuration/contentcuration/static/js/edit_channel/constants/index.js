const ContentKinds = require('./ContentKinds');
const FormatPresets = require('./FormatPresets');
const Languages = require('./Languages');
const Licenses = require('./Licenses');
const MasteryModels = require('./MasteryModels');
const Roles = require('./Roles');

module.exports = {
  ContentKinds,
  FormatPresets,
  Languages,
  Licenses,
  MasteryModels,
  Roles,
};
