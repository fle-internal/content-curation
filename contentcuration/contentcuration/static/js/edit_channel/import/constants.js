export const PageTypes = {
  TREE_VIEW: 'TREE_VIEW',
  IMPORT_PREVIEW: 'IMPORT_PREVIEW',
  SEARCH_RESULTS: 'SEARCH_RESULTS',
};
