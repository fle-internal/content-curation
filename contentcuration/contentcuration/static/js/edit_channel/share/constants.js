export const Permissions = {
  OWNER: 'manage',
  EDIT: 'edit',
  VIEW_ONLY: 'view'
};

export const PermissionRanks = [
	{
		"shareMode": Permissions.VIEW_ONLY,
		"rank": 0,
		"field": 'viewers'
	},
	{
		"shareMode": Permissions.EDIT,
		"rank": 1,
		"field": 'editors'
	},
	{
		"shareMode": Permissions.OWNER,
		"rank": 2,
		"field": null
	},
]
