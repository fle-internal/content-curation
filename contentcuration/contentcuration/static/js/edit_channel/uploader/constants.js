export const TabNames = {
  DETAILS: 'DETAILS',
  PREVIEW: 'PREVIEW',
  QUESTIONS: 'QUESTIONS',
  PREREQUISITES: 'PREREQUISITES',
};

export const modes = {
  EDIT: 'EDIT',
  NEW_TOPIC: 'NEW_TOPIC',
  NEW_EXERCISE: 'NEW_EXERCISE',
  UPLOAD: 'UPLOAD',
  VIEW_ONLY: 'VIEW_ONLY',
};

export const AssessmentItemToolbarActions = {
  MOVE_ITEM_UP: 'MOVE_ITEM_UP',
  MOVE_ITEM_DOWN: 'MOVE_ITEM_DOWN',
  EDIT_ITEM: 'EDIT_ITEM',
  ADD_ITEM_ABOVE: 'ADD_ITEM_ABOVE',
  ADD_ITEM_BELOW: 'ADD_ITEM_BELOW',
  DELETE_ITEM: 'DELETE_ITEM',
};

export const AssessmentItemValidationErrors = {
  BLANK_QUESTION: 'BLANK_QUESTION',
  INVALID_NUMBER_OF_CORRECT_ANSWERS: 'INVALID_NUMBER_OF_CORRECT_ANSWERS',
};

// should correspond to backend types
export const AssessmentItemTypes = {
  SINGLE_SELECTION: 'single_selection',
  MULTIPLE_SELECTION: 'multiple_selection',
  TRUE_FALSE: 'true_false',
  INPUT_QUESTION: 'input_question',
};

export const AssessmentItemTypeLabels = {
  [AssessmentItemTypes.SINGLE_SELECTION]: 'Single selection',
  [AssessmentItemTypes.MULTIPLE_SELECTION]: 'Multiple selection',
  [AssessmentItemTypes.TRUE_FALSE]: 'True/False',
  [AssessmentItemTypes.INPUT_QUESTION]: 'Input question',
};
