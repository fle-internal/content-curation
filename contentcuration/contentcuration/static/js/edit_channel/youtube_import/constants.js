export const PageTypes = {
  SUBMIT_URL: 'SUBMIT_URL',
  CONFIRM: 'CONFIRM',
};

export const ImportStatus = {
  IDLE: 'IDLE',
  EXTRACTING: 'EXTRACTING',
  EXTRACT_ERROR: 'EXTRACT_ERROR',
  DOWNLOADING: 'DOWNLOADING',
  DOWNLOAD_ERROR: 'DOWNLOAD_ERROR',
  START_IMPORT: 'START_IMPORT',
  DONE: 'DONE',
};
