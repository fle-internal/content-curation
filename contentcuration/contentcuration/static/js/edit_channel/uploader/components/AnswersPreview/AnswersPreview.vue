<template>
  <div>
    <div v-if="!answers || !answers.length">
      No answers yet
    </div>

    <template v-else>
      <template v-if="isSingleSelection || isTrueFalse">
        <VRadioGroup v-model="correctAnswersIndices">
          <VRadio
            v-for="(answer, idx) in answers"
            :key="idx"
            :label="answer.answer"
            :value="idx"
            disabled
          />
        </VRadioGroup>
      </template>

      <template v-if="isMultipleSelection">
        <VCheckbox
          v-for="(answer, idx) in answers"
          :key="idx"
          v-model="correctAnswersIndices"
          :label="answer.answer"
          :value="idx"
          disabled
        />
      </template>

      <ul v-if="isInputQuestion">
        <li
          v-for="(answer, idx) in answers"
          :key="idx"
        >
          {{ answer.answer }}
        </li>
      </ul>
    </template>
  </div>
</template>

<script>

  import { AssessmentItemTypes } from '../../constants';
  import { getCorrectAnswersIndices } from '../../utils';

  export default {
    name: 'AnswersPreview',
    props: {
      questionKind: {
        type: String,
        validator: value => {
          return Object.values(AssessmentItemTypes).includes(value);
        },
        default: AssessmentItemTypes.SINGLE_SELECTION,
      },
      answers: {
        type: Array,
      },
    },
    computed: {
      isSingleSelection() {
        return this.questionKind === AssessmentItemTypes.SINGLE_SELECTION;
      },
      isMultipleSelection() {
        return this.questionKind === AssessmentItemTypes.MULTIPLE_SELECTION;
      },
      isTrueFalse() {
        return this.questionKind === AssessmentItemTypes.TRUE_FALSE;
      },
      isInputQuestion() {
        return this.questionKind === AssessmentItemTypes.INPUT_QUESTION;
      },
      correctAnswersIndices() {
        return getCorrectAnswersIndices(this.questionKind, this.answers);
      },
    },
  };

</script>
