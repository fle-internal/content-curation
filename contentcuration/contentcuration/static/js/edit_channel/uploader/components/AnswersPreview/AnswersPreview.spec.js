import { shallowMount, mount } from '@vue/test-utils';

import { AssessmentItemTypes } from '../../constants';
import AnswersPreview from './AnswersPreview';

describe('AnswersPreview', () => {
  let wrapper;

  it('smoke test', () => {
    const wrapper = shallowMount(AnswersPreview);

    expect(wrapper.isVueInstance()).toBe(true);
  });

  describe('with no answers', () => {
    beforeEach(() => {
      wrapper = shallowMount(AnswersPreview, {
        propsData: {
          answers: [],
        },
      });
    });

    it('renders a placeholder', () => {
      expect(wrapper.html()).toContain('No answers yet');
    });
  });

  describe('for a single selection question', () => {
    beforeEach(() => {
      wrapper = mount(AnswersPreview, {
        propsData: {
          questionKind: AssessmentItemTypes.SINGLE_SELECTION,
          answers: [
            { answer: 'Mayonnaise (I mean you can, but...)', correct: true, order: 1 },
            { answer: 'Peanut butter', correct: false, order: 2 },
          ],
        },
      });
    });

    it('renders answers as disabled radio selects', () => {
      const inputs = wrapper.findAll('input');
      const labels = wrapper.findAll('label');

      expect(inputs.length).toBe(2);
      expect(labels.length).toBe(2);

      for (let n in [0, 1]) {
        expect(inputs.at(n).attributes()['type']).toBe('radio');
        expect(inputs.at(n).attributes()['disabled']).toBeTruthy();
      }

      expect(labels.at(0).text()).toBe('Mayonnaise (I mean you can, but...)');
      expect(labels.at(1).text()).toBe('Peanut butter');
    });

    it('renders a correct answer as checked', () => {
      const inputs = wrapper.findAll('input');

      expect(inputs.at(0).element.checked).toBe(true);
      expect(inputs.at(1).element.checked).toBe(false);
    });
  });

  describe('for a multiple selection question', () => {
    beforeEach(() => {
      wrapper = mount(AnswersPreview, {
        propsData: {
          questionKind: AssessmentItemTypes.MULTIPLE_SELECTION,
          answers: [
            { answer: 'Mayonnaise (I mean you can, but...)', correct: true, order: 1 },
            { answer: 'Peanut butter', correct: false, order: 2 },
            { answer: 'Jelly', correct: true, order: 3 },
          ],
        },
      });
    });

    it('renders answers as disabled checkboxes', () => {
      const inputs = wrapper.findAll('input');
      const labels = wrapper.findAll('label');

      expect(inputs.length).toBe(3);
      expect(labels.length).toBe(3);

      for (let n in [0, 1, 2]) {
        expect(inputs.at(n).attributes()['type']).toBe('checkbox');
        expect(inputs.at(n).attributes()['disabled']).toBeTruthy();
      }

      expect(labels.at(0).text()).toBe('Mayonnaise (I mean you can, but...)');
      expect(labels.at(1).text()).toBe('Peanut butter');
      expect(labels.at(2).text()).toBe('Jelly');
    });

    it('renders all correct answers as checked', () => {
      const inputs = wrapper.findAll('input');

      expect(inputs.at(0).element.checked).toBe(true);
      expect(inputs.at(1).element.checked).toBe(false);
      expect(inputs.at(2).element.checked).toBe(true);
    });
  });

  describe('for a true/false question', () => {
    beforeEach(() => {
      wrapper = mount(AnswersPreview, {
        propsData: {
          questionKind: AssessmentItemTypes.TRUE_FALSE,
          answers: [
            { answer: 'True', correct: false, order: 1 },
            { answer: 'False', correct: true, order: 2 },
          ],
        },
      });
    });

    it('renders answers as disabled radio selects', () => {
      const inputs = wrapper.findAll('input');
      const labels = wrapper.findAll('label');

      expect(inputs.length).toBe(2);
      expect(labels.length).toBe(2);

      for (let n in [0, 1]) {
        expect(inputs.at(n).attributes()['type']).toBe('radio');
        expect(inputs.at(n).attributes()['disabled']).toBeTruthy();
      }

      expect(labels.at(0).text()).toBe('True');
      expect(labels.at(1).text()).toBe('False');
    });

    it('renders a correct answer as checked', () => {
      const inputs = wrapper.findAll('input');

      expect(inputs.at(0).element.checked).toBe(false);
      expect(inputs.at(1).element.checked).toBe(true);
    });
  });

  describe('for an input question', () => {
    beforeEach(() => {
      wrapper = mount(AnswersPreview, {
        propsData: {
          questionKind: AssessmentItemTypes.INPUT_QUESTION,
          answers: [
            { answer: 'Mayonnaise (I mean you can, but...)', correct: true, order: 1 },
            { answer: 'Peanut butter', correct: true, order: 2 },
          ],
        },
      });
    });

    it('renders all answers', () => {
      expect(
        wrapper
          .findAll('li')
          .at(0)
          .text()
      ).toBe('Mayonnaise (I mean you can, but...)');
      expect(
        wrapper
          .findAll('li')
          .at(1)
          .text()
      ).toBe('Peanut butter');
    });
  });
});
