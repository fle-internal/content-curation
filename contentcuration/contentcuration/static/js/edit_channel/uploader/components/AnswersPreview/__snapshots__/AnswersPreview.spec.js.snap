// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswersPreview for an input question renders all answers 1`] = `
<div>
  <!---->
  <!---->
  <div role="list" class="v-list theme--light">
    <div role="listitem">
      <div class="v-list__tile theme--light">
        Mayonnaise (I mean you can, but...)
      </div>
    </div>
    <div role="listitem">
      <div class="v-list__tile theme--light">
        Peanut butter
      </div>
    </div>
  </div>
</div>
`;
